src/components/shared/Fields/InputStatus/HookForm/InputStatus.tsx

```diff
@@ -0,0 +1,59 @@
+import { useIntl } from 'react-intl';
+import React from 'react';
+
+import { getMainClasses } from '~utils/css';
+import { formatText } from '~utils/intl';
+import { isNil } from '~utils/lodash';
+import { HookFormInputProps } from '~shared/Fields/Input/HookForm';
+import { Message } from '~types';
+
+import styles from '../InputStatus.css';
+
+interface HookFormInputStatusProps
+  extends Pick<
+    HookFormInputProps,
+    'appearance' | 'status' | 'statusValues' | 'isLoading' | 'loadingAnnotation'
+  > {
+  /** Error text (if applicable) */
+  error?: Message;
+
+  /** Has input field been touched? */
+  touched?: boolean;
+}
+
+const displayName = 'HookFormInputStatus';
+
+const HookFormInputStatus = ({
+  appearance = {},
+  error,
+  isLoading,
+  loadingAnnotation = '',
+  status,
+  statusValues,
+  touched,
+}: HookFormInputStatusProps) => {
+  const { formatMessage } = useIntl();
+  const errorText = formatText(error);
+  const statusText = formatText(status, statusValues);
+  const loadingText = formatMessage(
+    { id: 'status.loading' },
+    { optionalText: formatText(loadingAnnotation) },
+  );
+  const text = errorText || statusText;
+  const Element = appearance.direction === 'horizontal' ? 'span' : 'p';
+  return (
+    <Element
+      className={getMainClasses(appearance, styles, {
+        error: !!error && !isLoading,
+        hidden:
+          (!text && !isLoading) || (!!error && !isNil(touched) && !touched),
+      })}
+    >
+      {isLoading ? loadingText : text}
```

Comment:
Bringing the loading state into this component means we don't have to think about it higher up the component hierarchy, we can just pass in an `isLoading` prop.